lib/IntermediateRepresentation/builder.h

```diff
@@ -0,0 +1,139 @@
+    return InsertTerminator(BranchStmt::Create(block));
+  }
+  Statement &CreateCall(const FunctionType *type, const Value *callee,
+      CallArguments &&arguments) {
+    return Insert(CallStmt::Create(type, callee, std::move(arguments)));
+  }
+  template<typename A>
+  Statement &CreateConditionalBranch(
+      A *condition, BasicBlock *trueBlock, BasicBlock *falseBlock) {
+    return InsertTerminator(
+        BranchStmt::Create(condition, trueBlock, falseBlock));
+  }
+  Statement &CreateDealloc(const Expression *object) {
+    return Insert(DeallocateStmt::Create(object));
+  }
+  template<typename A> Statement &CreateExpr(const A *a) {
+    return Insert(ExprStmt::Create(a));
+  }
+  Statement &CreateIOCall(
+      InputOutputCallType call, IOCallArguments &&arguments) {
+    return Insert(IORuntimeStmt::Create(call, std::move(arguments)));
+  }
+  Statement &CreateIndirectBr(
+      Variable *var, const std::vector<BasicBlock *> &potentials) {
+    return InsertTerminator(IndirectBrStmt::Create(var, potentials));
+  }
+  Statement &CreateNullify(const parser::NullifyStmt *statement) {
+    return Insert(DisassociateStmt::Create(statement));
+  }
+  Statement &CreatePointerAssign(const Expression *lhs, const Expression *rhs) {
+    return Insert(PointerAssignStmt::Create(lhs, rhs));
+  }
+  Statement &CreateRetVoid() { return InsertTerminator(ReturnStmt::Create()); }
+  template<typename A> Statement &CreateReturn(A *expr) {
+    return InsertTerminator(ReturnStmt::Create(expr));
+  }
+  Statement &CreateRuntimeCall(
+      RuntimeCallType call, RuntimeCallArguments &&arguments) {
+    return Insert(RuntimeStmt::Create(call, std::move(arguments)));
+  }
+  Statement &CreateSwitch(const Evaluation &condition, BasicBlock *defaultCase,
+      const SwitchStmt::ValueSuccPairListType &rest) {
+    return InsertTerminator(SwitchStmt::Create(condition, defaultCase, rest));
+  }
+  Statement &CreateSwitchCase(const Evaluation &condition,
+      BasicBlock *defaultCase,
+      const SwitchCaseStmt::ValueSuccPairListType &rest) {
+    return InsertTerminator(
+        SwitchCaseStmt::Create(condition, defaultCase, rest));
+  }
+  Statement &CreateSwitchType(const Evaluation &condition,
+      BasicBlock *defaultCase,
+      const SwitchTypeStmt::ValueSuccPairListType &rest) {
+    return InsertTerminator(
+        SwitchTypeStmt::Create(condition, defaultCase, rest));
+  }
+  Statement &CreateSwitchRank(const Evaluation &condition,
+      BasicBlock *defaultCase,
+      const SwitchRankStmt::ValueSuccPairListType &rest) {
+    return InsertTerminator(
+        SwitchRankStmt::Create(condition, defaultCase, rest));
+  }
+  Statement &CreateUnreachable() {
+    return InsertTerminator(UnreachableStmt::Create());
+  }
+
+  void PushBlock(BasicBlock *block) { blockStack_.push_back(block); }
+  BasicBlock *PopBlock() {
+    auto *block{blockStack_.back()};
+    blockStack_.pop_back();
+    return block;
+  }
+  void dump() const;
```

Comment:
If this member function took a `std::ostream &` argument, it would probably be best.

Reply:
OK. `dump` really isn't implemented yet.